core/src/main/java/org/apache/carbondata/core/datastore/compression/ZstdCompressor.java

```diff
@@ -74,4 +74,16 @@ public long maxCompressedLength(long inputSize) {
   public boolean supportUnsafe() {
     return false;
   }
+
+  @Override public int unCompressedLength(byte[] data, int offset, int length) {
+    throw new RuntimeException("Unsupported operation Exception");
+  }
+
+  @Override public int rawUncompress(byte[] data, int offset, int length, byte[] output) {
+    throw new RuntimeException("Unsupported operation Exception");
+  }
+
+  @Override public boolean supportReusableBuffer() {
+    return false;
```

Comment:
Move the default implementation to Abstract class `AbstractCompressor`

Reply:
ok